### ISSUE TITLE
V2 – Smoothness/Segments control on linear extrude

### ISSUE THREAD

**Issue by MathieuLoutre** [jscad/OpenJSCAD.org#681]
## Expected Behavior
When extruding a bezier path, we should have control over how many segments are created for the curves

## Actual Behavior
There seem to be a default and no options to change that

## Steps to Reproduce the Problem
Following the SVG import example, I've written this script which comverts an SVG into geometry, then converts the path2 into polygons/geom2 and then extrudes it (using `geom2` as target for the deserializer gives an out of memory when extruding). 

```
const svg = svgDeSerializer.deserialize({ output: 'geometry', target: 'path2' }, `<svg xmlns="http://www.w3.org/2000/svg"><path d="M249 108C271 168 292 260 305 403L249 403C243 338 234 259 219 194C201 255 181 330 167 403C213 435 259 471 259 575C259 703 190 708 175 708C136 708 85 674 85 557C85 522 90 473 98 419C54 391 13 354 13 238C13 97 66-2 142-2C158-2 185 1 214 39C231 13 250-3 270-3C279-3 290-2 297 1L302 54C275 53 259 84 249 108ZM146 582C146 629 153 651 171 651C193 651 201 617 201 575C201 512 183 480 157 458C150 503 146 545 146 582ZM145 65C93 65 76 193 76 249C76 299 89 330 109 354C127 262 153 164 184 94C173 76 160 65 145 65Z"/></svg>`)

const poly = svg.map((shape) => primitives.polygon({ points: shape.points }))

let shapes = booleans.union(extrusions.extrudeLinear({ height: 10 }, polyLetter))
shapes = transforms.center({}, transforms.mirrorX(transforms.mirrorY(shapes)))

const stl = Buffer.concat(stlSerializer.serialize({ binary: true }, shapes).map((buff) => Buffer.from(buff)))

fs.writeFileSync(`${glyph.name}.stl`, stl, 'binary')
```

Admittedly, I don't fully understand what's going on but in SCAD I can use $fn to control how smooth the curves are. Is there a way to do that in JSCAD? Where is this going wrong? (I suspect it's the conversion into polygon which might ignore curves but I'm not sure what else to convert it into). I also noticed an "extrudeAlongPath" in the repo but I'm not sure how to approach it.

## Specifications

  - Version: 2.0.0-alpha.4
  - Platform: OSX 10.14.6
  - Environment: Node.js

**Comment by z3dev** [comment]
Yeah. It seems that 'segments' is not available when asking for output: 'geometry'.

As for $fn, there's a new utility function called radiusToSegments(). This function will calculate the 'segments' from a target radius, etc.

I can think of only one work around. The SVG can be converted to JSCAD script. And the script can be tweaked to specify 'segments' for each of the polygons. Not the best but might work.

**Comment by MathieuLoutre** [comment]
Gotcha! I didn't think it would happen during the SVG import but now found that manually changing the default value of segments in `appendBezier` seems to do the trick. I've tried, briefly, to add a segments argument to `deserialize` and though I'm nearly there, I think I lost myself in the chain as I'm not familiar with the project structure yet. That said, it seems doable?

**Comment by z3dev** [comment]
Super! It should be possible.

You are welcome to submit changes via a pull request. Be sure to create a unique branch to work upon, and then submit a pull request to the V2 branch.

**Comment by MathieuLoutre** [comment]
I've worked something out but of course I'm still new to the codebase. Looking forward to your comments/tweaks!

**Comment by z3dev** [comment]
@MathieuLoutre thanks again for the changes.

if this issue has been resolved now then please close.

**Comment by MathieuLoutre** [comment]
Closed by #682
